template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateProcessor.java

```diff
@@ -243,7 +247,31 @@ private Map<String, Set<ServiceComponent>> getNonGatewayServicesByHostGroup() {
                                 .filter(i -> !i.getComponent().equalsIgnoreCase("GATEWAY"))
                                 .collect(toSet())
                 ));
+    }
+
+    public List<String> getUnhealthyDependentHostGroups(StackDto stackdto, Set<String> dependentComponent, String undefinedDependency) {
```

Comment:
So most of the subsequent logic moves out to UpdateNodeCountValidator.

Reply:
Adding to this. I've cherry-picked this commit to a different branch for downstream periscope changes, could this and its related logic be moved to the newly introduced `RoleDependencyService`, which is, in turn, invoked directly by `UpdateNodeCountValidator`? I find that to be a convenient location for future AutoscaleAPI related changes. @Vatsal23082000

Reply:
Okay, I will make this changes.

Reply:
@aakashshankar Should I add all the health check logic in UpdateNodeCountValidator? 
Because adding a new function for health check in RoledependencyService Would not make more sense as that service is exclusively giving us the dependent component.

Reply:
That service can be renamed to perform healthChecks as well since it's a new service that's introduced

Reply:
@aakashshankar I have updated the PR.